### PR TITLE
Feature 150 tests docker image

### LINKED ISSUE #150: Docker image for tests
As a developer, I would like to run tests as a docker image.

### PR DESCRIPTION
**Pull request by platonovsimeon**
Resolves #150

### CODE REVIEW

File: src/main/java/fi/metatavu/metamind/server/keycloak/AuthenticationController.java [hunk 1]
```diff
@@ -51,18 +63,52 @@
...
+  private static final String ADMIN_CLIENT_SECRET = getAdminClientSecret();
   private static final String AUTHORIZATION_EXCEPTION_MESSAGE = "Client representation is NULL";
   
+  private static final String API_CLIENT_ID = getApiClientId();
+  private static final String API_CLIENT_SECRET = getApiClientSecret();
+  
   @Inject 
   private static Logger logger;
   
+  private static final String getRealm() {
```
Comment (anttileppa) [inline src/main/java/fi/metatavu/metamind/server/keycloak/AuthenticationController.java]: Move private methods after public methods

Comment (anttileppa) [inline src/main/java/fi/metatavu/metamind/server/keycloak/AuthenticationController.java]: Add javadocs to all methods

File: src/main/java/fi/metatavu/metamind/server/keycloak/AuthenticationController.java [hunk 2]
```diff
@@ -285,6 +331,7 @@ public UUID getCreateResponseId(Response response) {
       Keycloak keycloak = getAdminClient();
       RealmResource realm = keycloak.realm(REALM);
       ClientRepresentation client = getClient(keycloak);
+      ClientRepresentation apiClient = getClient(getApiClient());
```
Comment (anttileppa) [inline src/main/java/fi/metatavu/metamind/server/keycloak/AuthenticationController.java]: could we retieve api client with admin client (keycloak)? This way we would not need to create two resource consuming clients

**Review by anttileppa — CHANGES_REQUESTED** [review]

File: src/main/java/fi/metatavu/metamind/server/keycloak/AuthenticationController.java (hunk 2 above)
Comment (platonovsimeon) [reply src/main/java/fi/metatavu/metamind/server/keycloak/AuthenticationController.java]: fixed

File: src/main/java/fi/metatavu/metamind/server/keycloak/AuthenticationController.java (hunk 1 above)
Comment (platonovsimeon) [reply src/main/java/fi/metatavu/metamind/server/keycloak/AuthenticationController.java]: fixed

Comment (platonovsimeon) [reply src/main/java/fi/metatavu/metamind/server/keycloak/AuthenticationController.java]: fixed

File: src/main/java/fi/metatavu/metamind/server/keycloak/AuthenticationController.java [hunk 3]
```diff
@@ -122,23 +88,21 @@ private static final String getApiClientSecret () {
...
     Set<ScopeRepresentation> scopeRepresentations = scopes.stream()
       .map(AuthorizationScope::getName)
       .map(ScopeRepresentation::new)
       .collect(Collectors.toSet());
     
     ResourceRepresentation resource = new ResourceRepresentation(name, scopeRepresentations, uri, type);
     resource.setOwner(ownerId.toString());
-    
-    Response createResponse = resources.create(resource);
+    resources.create(resource);
```
Comment (anttileppa) [inline src/main/java/fi/metatavu/metamind/server/keycloak/AuthenticationController.java]: Add check that this request responds with ok status

**Review by anttileppa — CHANGES_REQUESTED** [review]

**Pull request merged**